runtime/lua/vim/treesitter.lua

```diff
@@ -113,18 +113,26 @@ end
 ---
 --- If needed, this will create the parser.
 ---
+--- NOTE: This function will return `nil` rather than throwing an error in
+--- Nvim 0.12. Please pass `opts.error = false` to achieve this behavior
+--- now.
+---
 ---@param bufnr (integer|nil) Buffer the parser should be tied to (default: current buffer)
 ---@param lang (string|nil) Language of this parser (default: from buffer filetype)
 ---@param opts (table|nil) Options to pass to the created language tree
 ---
----@return vim.treesitter.LanguageTree object to use for parsing
+---@return vim.treesitter.LanguageTree? object to use for parsing
 function M.get_parser(bufnr, lang, opts)
   -- TODO(ribru17): Remove _get_parser and move that logic back here once the breaking function
   -- signature change is acceptable.
+  opts = opts or {}
+
   local parser = M._get_parser(bufnr, lang, opts)
-  if not parser then
+  local should_error = opts.error == nil or opts.error
+  if not parser and should_error then
     vim.notify_once(
-      'WARNING: vim.treesitter.get_parser will return nil instead of raising an error in Neovim 0.12',
+      'WARNING: vim.treesitter.get_parser will return nil instead of raising an error in Neovim'
+        .. ' 0.12. Please run with "opts.error = false".',
```

Comment:
I thought the point was to _remove_ the warning because it can't be slurped by `pcall`?

Reply:
Oh, I thought the point was the make the warning remov*able* by users (by passing the argument); if the warning is removed, will it not be difficult to know (as a user) that there will be a breaking change in the future?

Reply:
Careful; user != caller (that would be a plugin). The point here is that we need to be careful with deprecations that concern plugins. In particular, we should absolutely minimize the chance that the warning will trigger a "press ENTER" prompt.

Personally, I would consider a soft deprecation here (doc change only).

Reply:
Ok, that makes sense